src/main/java/io/knotx/junit5/util/RequestUtil.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (C) 2018 Knot.x Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.knotx.junit5.util;
+
+import io.reactivex.Observable;
+import io.reactivex.Single;
+import io.reactivex.functions.Consumer;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.junit5.VertxTestContext;
+import io.vertx.reactivex.core.http.HttpClient;
+import io.vertx.reactivex.core.http.HttpClientRequest;
+import io.vertx.reactivex.core.http.HttpClientResponse;
+
+public interface RequestUtil {
+  /**
+   * Generate reactivex async request for given resource parameters
+   *
+   * @param client Vert.x client
+   * @param method target HTTP method
+   * @param port target port
+   * @param domain target domain
+   * @param uri resource to request
+   * @param requestBuilder handler for request body and params
+   * @return reactivex wrapper
+   */
+  static Observable<HttpClientResponse> asyncRequest(
+      HttpClient client,
+      HttpMethod method,
+      int port,
+      String domain,
+      String uri,
+      java.util.function.Consumer<HttpClientRequest> requestBuilder) {
+    return Observable.unsafeCreate(
```

Comment:
You can wrap HttpClient with WebClient and do all this stuff RX-way.

Reply:
Due to its limited use (one occurrence in knotx-core) I think we can safely mark this as deprecated and throw it out of the window on next opportunity.

I'll move it back to KnotxTestUtils where it belongs, it was added to RequestUtil under assumption that it was more widely used.